crates/wac-types/src/aggregator.rs

```diff
@@ -243,32 +226,22 @@ impl TypeAggregator {
                 // Otherwise, the source *must* be a subtype of the target
                 // We'll remap the source below and replace
                 checker
-                    .is_subtype(
-                        *source_kind,
-                        types,
-                        target_kind,
-                        &self.types,
-                        SubtypeCheck::Contravariant,
-                    )
+                    .is_subtype(*source_kind, types, target_kind, &self.types)
                     .with_context(|| format!("mismatched type for export `{name}`"))?;
             }
 
             let remapped = self.remap_item_kind(types, *source_kind, checker)?;
             self.types[existing].imports.insert(name.clone(), remapped);
         }
 
+        checker.revert();
+
```

Comment:
Can you explain why this stateful approach is better than passing whether the check is covariant or contravariant as an argument?

Reply:
The current check doesn't properly account for the inversion that takes place when these things nest; it was always forcing the check at each recursion point to a particular kind of check when rather it should invert the current relationships.

For example, if we're type checking an instance import of a world, the type check that's performed on the instance's exports should be *contravariant*, but because the subtype checks were passing *covariant*, the error messages in some of the test baselines came out backwards with regards to what was expected and what was found.

The current check was just one of those "refactor and attempt to improve, but forget why it was done that way originally" type of things.